src/NHibernate/Linq/LinqExtensionMethodAttribute.cs

```diff
@@ -1,18 +1,34 @@
-using System;
-
 namespace NHibernate.Linq
 {
-	public class LinqExtensionMethodAttribute: Attribute
+	/// <summary>
+	/// Flag a method as being a SQL function call for the linq-to-nhibernate provider. Its
+	/// parameters will be used as the function call parameters. Unless <see cref="DBOnlyAttribute.DBOnly"/>
+	/// is explicitly setted to <see langword="true" />, the provider will execute the method when its does
+	/// arguments do not depend on the queried data.
+	/// </summary>
+	public class LinqExtensionMethodAttribute: DBOnlyAttribute
 	{
+		/// <summary>
+		/// Default constructor. The method call will be translated by the linq provider to
+		/// a function call having the same name than the method.
+		/// </summary>
 		public LinqExtensionMethodAttribute()
 		{
+			DBOnly = false;
```

Comment:
It's not true. Any static method regardless accepting the argument from the query or not is tried be evaluated.